src/Symfony/Component/DependencyInjection/Config/AutowireServiceResource.php

```diff
@@ -32,7 +33,21 @@ public function isFresh($timestamp)
             return false;
         }
 
-        return @filemtime($this->filePath) <= $timestamp;
+        // has the file *not* been modified? Definitely fresh
+        if (@filemtime($this->filePath) <= $timestamp) {
+            return true;
+        }
+
+        try {
+            $reflectionClass = new \ReflectionClass($this->class);
+        } catch (\ReflectionException $e) {
+            // the class does not exist anymore!
+            return false;
+        }
+
+        $newSelf = AutowirePass::createResourceForClass($reflectionClass);
+
+        return $this == $newSelf;
```

Comment:
Now fixed.

@weaverryan originally the PR was about fixing a test where this function should return false.
In PHP, when comparing object using strict `===` it returns `true` if this is the same instance of the class, in that case it would always be `false`, so the test would always pass.
We could use `==` to compare the objects since it returns `true` for different instances with identical properties, this is the same as casting them to arrays being strict, without checking the class.